rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,120 @@
+
+With the example terraform, `create-network.sh` can now use the `coda-network` tool (aka carey's key service, or the "Testnet SDK") to generate all of the neccessary keypairs for you, as well as the genesis ledger itself. In the initial implementation, this will be handled by the already ready to use but fairly hacky scripts `generate-keys-and-ledger.sh`, but as the coda-network tool becomes more fully featured then the sed+jq steps will be removed and this script will interract directly with coda-network.
+
+The flag `--bake-image=true` will cause the tool to also run `./scripts/bake-image.sh` to build a new docker image with all of the required network configuration and s3-cached files baked-in.
+
+Additional flags can be provided to the create-network command for changing the quantities of fish, whales, seeds, bots, etc. These quantaties are provided at this stage so that the genesis ledger can be populated with proper accounts for these services.
+
+Once the network has been defined and set up, you can always make manual changes to the `.tfvars` to customize the deployment. Additionally, `./scripts/bake-image.sh` can be run at any time to re-package a new daemon image with the required genesis ledger, proof, and configuration.  Any future edits to the keypairs, keysets, or the genesis ledger can be performed directly with coda-network.
+
+### ./scripts/bake-image.sh
+
+The `./scripts/bake-image.sh <testnet_name> --docker-image <image>` script is a simple entrypoint for building a new image based on the provided one, but with the genesis ledger, proofs, and proving keys baked-in to reduce network startup time at the expense of this one-time step (Pain Point #4). If no --docker-image flag is specified, the script will use the docker_image variable in `<testnet_name>.tfvars`. This can be run on any existing network, and will produce a docker image with the tag `gcr.io/o1labs-192920/mina-daemon:<image>-baked-<sha256>` where sha256 is the hash of the genesis_ledger for `<testnet_name>`. The script will also insert the above image tag into the `baked_docker_image` field in `<testnet_name>.tfvars`. This `baked_docker_image` parameterr will allow the terraform to assume the genesis ledger is already included in the image and will not use a runtimeConfig to read the genesis_ledger. Using pre-baked images allows for all deployments to skip the generate-proof stage and avoids duplicate work of downloading and unpacking keys from s3, but this script should still make it easy to iterate on a network deployment when images have bugs and need to be updated.
```

Comment:
Mind explicitly enumerating this list? Maybe some sort of mapping between today->where we're going with this proposal in terms of these artifacts? We discussed for sure but thinking it will help clarify things a bit.